README.md

```diff
@@ -0,0 +1,91 @@
+## What is `copr-test`
+
+The `copr-test` is the integration test for `Coprocessor` of the TiKV. The `Coprocessor` is used to execute TiDB push-down
+executor to improve the database performance. It's important to keep execution result consistent between TiDB and `Coprocessor`.
+This is the purpose of `copr-test`.
+
+The basic principle of `copr-test` is to execute test cases in TiDB standalone (with `mocktikv`) and TiDB cluster (with `Coprocessor`)
+and compare the execution result. We think the TiDB and Coprocessor is consistent if the results in both sides are same, vice versa.
+
+The `Coprocessor` contains two execution frameworks, `Batch` (vectorized execution engine) and `Non-Batch` in current
+stage (remove the `Non-Batch` version is in plan). So all test cases should be passed in the following execution environment.
+
+1. TiDB with mocktikv
+2. TiDB + TiKV with Non-Batch
+3. TiDB + TiKV with Batch
+
+### How do I know which expressions will be pushed down?
+
+The `Coprocessor` does not implement all builtin functions (or implemented but not tested fully) which have implemented by TiDB,
+so we can only push down the functions that Coprocessor has implemented and fully tested. Whether a function is pushed down or not
+is determined by the function `canFuncBePushed` in TiDB [tidb-expression/expr_to_pb.go](https://github.com/pingcap/tidb/blob/a090e6be2991bf85b18fcdb096f84d41f4f6bd85/expression/expr_to_pb.go#L303)
+
+We have added a failpoint `PushDownTestSwitcher` in the function `canFuncBePushed` to hijack our customized push down condition.
+And we will push down all functions in our integration tests using `export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"all\")"`, see the `run_tests.sh`[./push-down-test/run-tests.sh].
+
+## How to add test cases?
+
+- Test cases location
+    
+    All test cases should be placed in `push-down-test/sql` directory, and all SQL files will be executed in lexical order by file name.
```

Comment:
we can put the sql file in sub dir of `push-down-test/sql`, to originate them in different dir.